### DISCUSSION TITLE
Integration with native side -instrumentation test file

### DISCUSSION THREAD

**Discussion by proximilian** [leancodepl/patrol#994] [Q&A] [answered]
Hello Guys,
I am trying to set up the Patrol's native automation feature by creating the instrumentation test file: `MainActivityTest.java` as described in Patrol's documentation (https://patrol.leancode.co/native/setup). Unfortunately my app is not bound with Java and uses `Kotlin`. My questions are: Should I create similar instrumentation file but `Kotlin` one or creation of java file is a must? 
Btw  _android/app/src/androidTest_ directory is missing in my project-> Is it necessary to create it or I can place the instrumentation file at _android/app/src/main/kotlin/..._ directory (as in my case)?
Please help, guys. I will appreciate it. 
Patrol seems to be great, so I can't help myself start using it.

**Comment by bartekpacia** [comment, marked as answer]
Hi @proximilian, sorry for being late :)

You have to use Java. I tried Kotlin but for some reason it doesn't work. Anyway, it's very little Java :)

> Btw `android/app/src/androidTest` directory is missing in my project. Is it necessary to create it?

Yes, you have to create it.

**Reply by artemperegut** [reply]
Sorry for bumping a closed discussion.

I tried using Kotlin and it seems to work.
```
import androidx.test.platform.app.InstrumentationRegistry
import io.qameta.allure.android.rules.LogcatRule
import io.qameta.allure.android.rules.ScreenshotRule
import io.qameta.allure.android.rules.WindowHierarchyRule
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized
import org.junit.runners.Parameterized.Parameters
import pl.leancode.patrol.PatrolJUnitRunner

@RunWith(Parameterized::class)
class MainActivityTest(private val dartTestName: String) {

    @get:Rule
    public val screenshotRule: ScreenshotRule = ScreenshotRule(ScreenshotRule.Mode.END, "ss_end")

    @get:Rule
    public val windowHierarchyRule: WindowHierarchyRule = WindowHierarchyRule()

    @get:Rule
    public val logcatRule: LogcatRule = LogcatRule()

    @Test
    fun runDartTest() {
        val instrumentation: PatrolJUnitRunner =
            InstrumentationRegistry.getInstrumentation() as PatrolJUnitRunner
        instrumentation.runDartTest(dartTestName)
    }

    companion object {
        @JvmStatic
        @Parameters(name = "{0}")
        fun testCases(): Array<out Any>? {
            val instrumentation: PatrolJUnitRunner =
                InstrumentationRegistry.getInstrumentation() as PatrolJUnitRunner
            instrumentation.setUp(MainActivity::class.java)
            instrumentation.waitForPatrolAppService()
            return instrumentation.listDartTests()
        }
    }
}
```

Or I missing something?
